app/uk/gov/hmrc/helptosavefrontend/health/HealthCheck.scala

```diff
@@ -0,0 +1,255 @@
+ * [[uk.gov.hmrc.helptosavefrontend.health.HealthCheck.PerformTest]] message and respond with a [[HealthCheckResult]].
+ * The appropriate interface of the children is handily provided by the [[HealthCheckRunner]] trait. After a
+ * configured number of successful tests the [[HealthCheck]] actor will switch tests by using the next [[Props]] in
+ * `runnerProps`. When the last [[Props]] in `runnerProps` is used, the next switch will result in the first
+ * [[Props]] being used again and the [[Props]] are iterated through again in a cyclical fashion.
+ *
+ * The [[HealthCheck]] actor will handle [[HealthCheckResult]]s by keeping a record of the number of failures using
+ * the given [[Metrics]] and triggering pager duty alerts when a configured number of consecutive failures
+ * has occurred.
+ *
+ * @param name The name of the test
+ * @param config The config variables will be looked for under the namespace `health-check.{name}` where `name` is
+ *               the name of the test
+ * @param scheduler      The scheduler used to schedule the health tests
+ * @param metrics        The metrics to be updated
+ * @param pagerDutyAlert A function which will alert pager duty
+ * @param runnerProps    The props of the tests to be run
+ */
+class HealthCheck(name:             String,
+                  config:           Config,
+                  scheduler:        Scheduler,
+                  metrics:          Metrics,
+                  pagerDutyAlert:   () ⇒ Unit,
+                  pagerDutyResolve: () ⇒ Unit,
+                  runnerProps:      NonEmptyVector[Props])
+  extends Actor with ActorLogging {
+
+  import uk.gov.hmrc.helptosavefrontend.health.HealthCheck._
+
+  import context.dispatcher
+
+  val minimumTimeBetweenTests: FiniteDuration =
+    config.get[FiniteDuration](s"health.$name.minimum-poll-period").value
+
+  val timeBetweenTests: FiniteDuration =
+    config.get[FiniteDuration](s"health.$name.poll-period").value.max(minimumTimeBetweenTests)
+
+  val testTimeout: FiniteDuration =
+    config.get[FiniteDuration](s"health.$name.poll-timeout").value
+
+  val numberOfTestsBetweenUpdates: Int =
+    config.get[Int](s"health.$name.poll-count-between-updates").value
+
+  val maximumConsecutiveFailures: Int =
+    config.get[Int](s"health.$name.poll-count-failures-to-alert").value
+
+  val numberOfTestsBetweenAlerts: Int =
+    config.get[Int](s"health.$name.poll-count-between-pager-duty-alerts").value
+
+  val failureHistogram: Histogram = metrics.defaultRegistry.histogram(s"health.$name.number-of-failures")
+
+  val propsQueue: NonEmptyCyclicalQueue[Props] = NonEmptyCyclicalQueue(runnerProps)
+
+  var currentRunnerProps: Props = propsQueue.next()
+
+  var performTestTask: Option[Cancellable] = None
+
+  override def receive: Receive = ok(0)
+
+  def ok(count: Int): Receive = performTest orElse {
+    case HealthCheckResult.Success(nanos) ⇒
+      log.info(s"Health check $name OK ${timeString(nanos)}")
```

Comment:
Can we start all comments wth say 'HealthCheck' allowing us to search for them in Kibana by ^HealthCheck

Reply:
yeah, that makes sense